src/game/common/rts/player.cpp

```diff
@@ -100,25 +120,476 @@ Player::~Player()
+    captainslog_dbgassert(m_playerTeamPrototypes.size() == 0, "Player::m_playerTeamPrototypes is not empty at game start!");
+    m_skillPointsModifier = 1.0f;
+    m_lastAttackedByFrame = 0;
+    m_playerIsPreorder = 0;
+    m_playerIsDead = 0;
+    m_radarCount = 0;
+    m_disableProofRadarCount = 0;
+    m_radarDisabled = 0;
+    m_activeBattlePlans[0] = 0;
+    m_activeBattlePlans[1] = 0;
+    m_activeBattlePlans[2] = 0;
+
+    if (m_battlePlanBonuses != nullptr) {
+        m_battlePlanBonuses->Delete_Instance();
+        m_battlePlanBonuses = nullptr;
+    }
+
+    Delete_Upgrade_List();
+    m_energy.Init(this);
+    m_missionStats.Init();
+
+    if (m_buildListInfo != nullptr) {
+        m_buildListInfo->Delete_Instance();
+        m_buildListInfo = nullptr;
+    }
+
+    m_defaultTeam = nullptr;
+
+    if (m_ai != nullptr) {
+        m_ai->Delete_Instance();
+        m_ai = nullptr;
+    }
+
+    if (m_resourceGatheringManager != nullptr) {
+        m_resourceGatheringManager->Delete_Instance();
+        m_resourceGatheringManager = nullptr;
+    }
+
+    for (int i = 0; i < SQUAD_COUNT; i++) {
+        if (m_squads[i] != nullptr) {
+            m_squads[i]->Delete_Instance();
+            m_squads[i] = nullptr;
+        }
+
+        m_squads[i] = new Squad();
+    }
+
+    if (m_aiSquad != nullptr) {
+        m_aiSquad->Delete_Instance();
+        m_aiSquad = nullptr;
+    }
+
+    m_aiSquad = new Squad();
+
+    if (m_tunnelSystem != nullptr) {
+        m_tunnelSystem->Delete_Instance();
+        m_tunnelSystem = nullptr;
+    }
+
+    m_canBuildBase = true;
+    m_canBuildUnits = true;
+    m_playerIsObserver = false;
+    m_bountyCostToBuild = 0.0f;
+    m_listInScoreScreen = true;
+    m_unitsShouldHunt = false;
+#ifdef GAME_DEBUG_STRUCTS
+    m_ignorePrereqs = false;
+    m_freeBuild = false;
+    m_instantBuild = false;
 #endif
+
+    if (pt != nullptr) {
+        m_side = pt->Get_Side_Name();
+        m_baseSide = pt->Get_Base_Side();
+        m_productionCostChanges = *pt->Get_Production_Cost_Changes();
+        m_productionTimeChanges = *pt->Get_Production_Time_Changes();
+        m_productionVeterancyLevels = *pt->Get_Production_Veterancy_Levels();
+        m_playerColor = pt->Get_Preferred_Color()->Get_As_Int() | 0xFF000000;
+        m_playerNightColor = m_playerColor;
+        m_money = *pt->Get_Money();
+        m_money.Set_Player_Index(Get_Player_Index());
+        m_handicap = *pt->Get_Handicap();
+
+        if (m_money.Get() == 0) {
+            if (g_theGameInfo != nullptr) {
+                m_money = *g_theGameInfo->Get_Money();
+            } else {
+                m_money = g_theWriteableGlobalData->m_defaultStartingCash;
+            }
+        }
+
+        m_playerDisplayName.Clear();
+        m_playerName.Clear();
+        m_playerNameKey = NAMEKEY_INVALID;
+        m_playerType = PLAYER_COMPUTER;
+        m_playerIsObserver = pt->Is_Observer();
+        m_playerIsDead = m_playerIsObserver;
```

Comment:
This means if player is not observer, then he is dead? Is this correct?

Reply:
Yes this is correct.